docs/agent-api.asciidoc

```diff
@@ -104,7 +104,7 @@ containing the data about to be sent to the APM Server.
 
 The format of the payload depends on the event type being sent.
 For details about the different formats,
-see the {apm-server-ref-v}/intake-api.html[APM Server intake API documentation].
+see the {apm-guide-ref}/intake-api.html[APM Server intake API documentation].
```

Comment:
https://github.com/elastic/observability-docs/issues/1385 says

`{apm-server-ref}/intake-api.html` -> `{apm-guide-ref}/api.html`

Though perhaps for this one we'd (eventually?) want https://www.elastic.co/guide/en/apm/guide/7.17/api-events.html ?  I'm not sure if the intent of this PR is to just fix the `{apm-server-ref-v}` reference, or to also change from the "Legacy APM Server Reference" to the new Agent-based APM server docs.

Reply:
Unfortunately, the goal here is just to fix the outdated `{apm-server-ref-v}` refs. There are improvements that could be made to all of the agent docs to better differentiate between running APM Server in standalone mode vs the APM integration, but I don't have the bandwidth to tackle those right now.

As for the API link. It looks like this was a mistake, but I'd agree that the current link is better than what I had proposed in elastic/observability-docs#1385.